docs/schematypes.jade

```diff
@@ -213,6 +213,37 @@ block content
     new Person({ name: { foo: 42 } }).name; 
     ```
 
+    <h4 id="numbers">Number</h4>
+
+    To declare a path as a number, you may use either the `Number` global
+    constructor or the string `'Number'`.
+
+    ```javascript
+    const schema1 = new Schema({ age: Number }); // age will be cast to a Number
+    const schema2 = new Schema({ age: 'Number' }); // Equivalent
+
+    const Car = mongoose.model('Car', schema2);
+    ```
+
+    There are several types of values that will be successfully cast to a Number.
+
+    ```javascript
+    new Car({ age: '15' }).age; // 15 as a Number
+    new Car({ age: true }).age; // 1 as a Number
+    new Car({ age: false }).age; // 0 as a Number
+    new Car({ age: { valueOf: () => 83 } }).age; // 83 as a Number
+    ```
+
+    If you pass an object with a `valueOf()` function that returns a Number, Mongoose will
+    call it and assign the returned value to the path.
+
+    There are a few noteworthy values that will not be cast to a Number:
+    The values `null` and `undefined` are not cast, they are instead assigned to the path. 
```

Comment:
Fine for now, but in general 'X is not cast' and 'Y are assigned' are tech writing antipatterns because they do not make it clear what does the casting and assigning. Whenever I read these type of sentencnes in docs, I'm always scratching my head as to whether I, the consumer of the software, am responsible for casting or whether Mongoose does it for me. Whenever you find yourself tempted to write something like this, just say "Mongoose does not cast the values `null` and `undefined`".